client/rpc/src/state/state_full.rs

```diff
@@ -122,7 +123,10 @@ impl<BE, Block: BlockT, Client> FullState<BE, Block, Client>
 			.max_key_changes_range(from_number, BlockId::Hash(to_meta.hash))
 			.map_err(client_err)?;
 		let filtered_range_begin = changes_trie_range
-			.map(|(begin, _)| (begin - from_number).saturated_into::<usize>());
+			.and_then(|(begin, _)| {
+				// avoids weird corner case where begin < from_number
```

Comment:
Can you elaborate when it happens? Why would `begin` ever be less than `from_number`?

Reply:
I was testing it yesterday and ran into this edge case when I passed the hash for the first block in the chain to `query_storage_at`, and `max_key_changes_range` returned `0` for `begin`.

Reply:
ping @tomusdrw, could you please take a look at this again?

Reply:
Thanks for explanations, I'd change the wording though, cause `weird` sounds like we don't know what's happening.

```suggestion
				// avoids a corner case where begin < from_number (happens when querying genesis)
```

Reply:
> I was testing it yesterday and ran into this edge case when I passed the hash for the first block in the chain to `query_storage_at`, and `max_key_changes_range` returned `0` for `begin`.

Can you add a test for this?